src/compiler/target/golang/mod.rs

```diff
@@ -0,0 +1,268 @@
+use super::*;
+
+pub struct Go {}
+
+impl Go {
+  /// Generate golang code using tokens from parser
+  pub fn generate(lb: &mut LangBuilder, t: AnilizedTokens) -> Result<(), LocationError> {
+
+    // TODO: Replace when file importing is implemented
+    lb.code("package main");
+
+    let mut code = Self {};
+
+    // define functions
+    for (_, func) in t.functions {
+      code.function(func, lb);
+    }
+
+    // define types
+    for (_, type_) in t.types {
+      code.code_type(type_, lb);
+    }
+
+    // define structs
+    for (_, structure) in t.structs {
+      code.structure(structure, lb);
+    }
+    
+    // define globals
+    for (_, glob) in t.vars {
+      code.global_var(glob, lb);
+    }
+
+    Ok(())
+  }
+  /// Get the type as a string
+  pub fn get_type(&mut self, type_: Option<tokenize::types::Type>) -> String {
+    match type_ {
+      Some(res) => {
+        match res.type_ {
+          tokenize::types::TypeType::Char => "rune".to_string(),
+          tokenize::types::TypeType::String => "string".to_string(),
+          tokenize::types::TypeType::Struct(res) => self.type_struct(res),
+          tokenize::types::TypeType::Int => "int".to_string(),
+          tokenize::types::TypeType::I8 => "int8".to_string(),
+          tokenize::types::TypeType::I16 => "int16".to_string(),
+          tokenize::types::TypeType::I32 => "int32".to_string(),
+          tokenize::types::TypeType::I64 => "int64".to_string(),
+          tokenize::types::TypeType::UInt => "uint".to_string(),
+          tokenize::types::TypeType::U8 => "uint8".to_string(),
+          tokenize::types::TypeType::U16 => "uint16".to_string(),
+          tokenize::types::TypeType::U32 => "uint32".to_string(),
+          tokenize::types::TypeType::U64 => "uint64".to_string(),
+          tokenize::types::TypeType::TypeRef(res) => res,
+          _ => unimplemented!()
```

Comment:
I would prefer to have every uninpmented item listed here so it's supper easy to know what we still have todo.